Client/src/Components/Graph/GraphView.tsx

```diff
@@ -1,235 +1,48 @@
-    const filteredDataset = completeDatasets.filter(dataset => dataset.id !== datasetId)
-    setCompleteDatasets(filteredDataset)
-    handleVariablesSelected(filteredDataset)
-    calculateExtremeBoundaries(filteredDataset)
-  }
-
-  const handleDatasetsSelected = (selectedDatasets: IDatasetModel[]) => {
-    const notYetSelectedDatasets: IDatasetModel[] = selectedDatasets.filter(selectedDataset => !isInStateAlready(selectedDataset))
-
-    const mergedDatasets: IDatasetModel[] = [...completeDatasets]
-    notYetSelectedDatasets.forEach(dataset => {
-      mergedDatasets.push(dataset)
-    })
-    handleVariablesSelected(mergedDatasets)
-    setCompleteDatasets(mergedDatasets)
-    calculateExtremeBoundaries(mergedDatasets)
-    handleClose()
+  const handleGraphStateChanged = (graphState: IGraphStateModel, completeDatasets: IDatasetModel[]) => {
+    const graphDatasets = getGraphDatasets(completeDatasets, graphState)
+    setGraphState(graphState)
+    setGraphDatasets(graphDatasets)
   }
 
-  const handleVariablesSelected = (selectedDatasets: IDatasetModel[]) => {
-    const variableNames = []
-    selectedDatasets.forEach(dataset => {
-      dataset.data.variables.forEach(variable => {
-        variableNames.push(variable.name)
-      })
-    })
-    const uniqueVariables = Array.from(new Set(variableNames))
-    setVariables(uniqueVariables)
-  }
-
-  const isInStateAlready = (dataset: IDatasetModel) => {
-    return completeDatasets.findIndex(existingDataset => existingDataset.id === dataset.id) != -1
-  }
-
-  const changeAxes = (newAxes: IVariableAndUnitModel) => {
-    setAxes(newAxes)
-  }
-
-  const calculateExtremeBoundaries = (datasets: IDatasetModel[]) => {
-    let minX = 9000, maxX = 0, minY = 9000, maxY = 0
-
-    const datalist: any[] = []
-    //todo refactor graph else this breaks
-    // datasets.forEach(dataset => datalist.push(dataset.points))
-    datalist.forEach(dataset => {
-      dataset.forEach(point => {
-        if (point.x > maxX) {
-          maxX = point.x
-        }
-        if (point.x < minX) {
-          minX = point.x
-        }
-        if (point.y > maxY) {
-          maxY = point.y
-        }
-        if (point.y < minY) {
-          minY = point.y
-        }
-      })
-    })
-    const extremeBoundaries: IDataPointExtremes = { minX: minX, maxX: maxX, minY: minY, maxY: maxY }
-    setDatasetBoundaries(extremeBoundaries)
-  }
-
-
   return (
     <>
-      <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"></link>
-      <h2>Graphs</h2>
-      <Modal
-        open={openModal}
-        onClose={handleClose}
-        className={classStyles().modalsearch}
-      >
-        <Box m={5} mb={10} width='100%'>
-          <Grid item xs={12}>
-            <Paper elevation={3}>
-              <Box m={0} p={2}>
-                <CancelIcon color="primary" className={classStyles().closeButton} width="2%" onClick={handleClose} />

```

Comment:
You removed it here, make sure to check next time to be aware you did not remove someone elses work